kaldb/src/main/java/com/slack/kaldb/metadata/schema/LuceneFieldDef.java

```diff
@@ -0,0 +1,82 @@
+package com.slack.kaldb.metadata.schema;
+
+import com.google.common.base.Objects;
+import com.slack.kaldb.logstore.InvalidFieldDefException;
+import com.slack.kaldb.metadata.core.KaldbMetadata;
+
+/*
+ * LuceneFieldDef describes the configs that can be set on a lucene field. This config defines how a field is indexed.
+ */
+public class LuceneFieldDef extends KaldbMetadata {
+  public final FieldType fieldType;
+  public final boolean isStored;
+  public final boolean isIndexed;
+  public final boolean isAnalyzed;
+  public final boolean storeNumericDocValue;
+
+  public LuceneFieldDef(
+      String name,
+      String fieldType,
+      boolean isStored,
+      boolean isIndexed,
+      boolean isAnalyzed,
+      boolean storeNumericDocValue) {
+    super(name);
+    this.fieldType = FieldType.valueOf(fieldType.trim().toUpperCase());
+    if (isAnalyzed && !isIndexed) {
+      throw new InvalidFieldDefException("Cannot set isAnalyzed without setting isIndexed");
```

Comment:
This should be using `Preconditions` checks, (ie, `checkArgument`), as that is the pattern used for all of the other `KaldbMetadata` classes.